lib/reek/spec/should_reek_only_of.rb

```diff
@@ -15,25 +15,30 @@ def matches?(actual)
       def matches_examiner?(examiner)
         @examiner = examiner
         @warnings = @examiner.smells
-        @warnings.length == 1 && @warnings[0].matches?(@klass, @patterns)
+        return false unless @warnings.length == 1
+        @warning = @warnings[0]
+        @warning.matches?(@smell_category, @smell_details)
```

Comment:
Good catch!
However we still need a variation of 

>> return false unless @warnings.length == 1

namely:

>> return false if @warnings.empty?

otherwise this matcher will return true on an empty smell list.

The other suggestion I incorporated though.